server/sonar-server/src/main/java/org/sonar/server/permission/ws/template/DeleteTemplateAction.java

```diff
@@ -76,21 +78,50 @@ private void doHandle(DeleteTemplateWsRequest request) {
         request.getTemplateId(), request.getOrganization(), request.getTemplateName()));
       checkGlobalAdmin(userSession, template.getOrganizationUuid());
 
-      checkTemplateUuidIsNotDefault(template.getUuid());
+      DefaultTemplates defaultTemplates = retrieveDefaultTemplates(dbSession, template);
+
+      checkTemplateUuidIsNotDefault(template, defaultTemplates);
       dbClient.permissionTemplateDao().deleteById(dbSession, template.getId());
+      updateViewDefaultTemplateWhenGovernanceIsNotInstalled(dbSession, template, defaultTemplates);
+
       dbSession.commit();
     }
   }
 
+  /**
+   * The default template for view can be removed when Governance is not installed. To avoid keeping a reference
+   * to a non existing template, we update the default templates.
+   */
+  private void updateViewDefaultTemplateWhenGovernanceIsNotInstalled(DbSession dbSession, PermissionTemplateDto template, DefaultTemplates defaultTemplates) {
```

Comment:
when Governance is not installed, the permission template referenced as the default template for views can be deleted. Not only does the WS allow it, it's not even visible in the UI that the template is the default for views. So, it's important to update the default template for views and set it to null so that we don't keep a reference to a non existing permission template